### ISSUE TITLE
Fix warning of lr schedulers

### ISSUE THREAD

**Issue by stephenroller** [facebookresearch/ParlAI#2874]
We need the lr_schedulers to switch to `step()` instead of `step(epoch)`, to get rid of a recent warning.

Migrate the code to have the `load_state_dict` of the parlai lr_scheduler call `load_state_dict({'last_epoch': self._num_training_updates})`.

Add tests for resuming an lr_scheduler. (This can be done as simply as making a new LR scheduler, calling newlr.load_state_dict(oldlr.state_dict()) and verifying the correct values on the next step())
